browser/resources/settings/brave_leo_assistant_page/brave_leo_assistant_page.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import {PolymerElement} from 'chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js';
+import {WebUiListenerMixin} from 'chrome://resources/cr_elements/web_ui_listener_mixin.js';
+import {PrefsMixin} from 'chrome://resources/cr_components/settings_prefs/prefs_mixin.js';
+import {I18nMixin} from 'chrome://resources/cr_elements/i18n_mixin.js';
+import {getTemplate} from './brave_leo_assistant_page.html.js'
+import {BraveLeoAssistantBrowserProxy, BraveLeoAssistantBrowserProxyImpl}
+  from './brave_leo_assistant_browser_proxy.js'
+
+
+const BraveLeoAssistantPageBase =
+  WebUiListenerMixin(I18nMixin(PrefsMixin(PolymerElement)))
+
+/**
+ * 'settings-brave-leo-assistant-page' is the settings page containing
+ * brave's Leo Assistant features.
+ */
+class BraveLeoAssistantPageElement extends BraveLeoAssistantPageBase {
+    static get is() {
+        return 'settings-brave-leo-assistant-page'
+    }
+
+    static get template() {
+        return getTemplate()
+    }
+
+    static get properties() {
+      return {
+        leoAssistantShowOnToolbarPref_: {
+          type: Boolean,
+          value: false,
+          notify: true,
+        },
+      }
+    }
+
+    leoAssistantShowOnToolbarPref_: boolean
+    leoAssistantShowPromptsPref_: boolean
+    browserProxy_: BraveLeoAssistantBrowserProxy =
+      BraveLeoAssistantBrowserProxyImpl.getInstance()
+
+    onResetAssistanceData_() {
+      this.browserProxy_.reset().then((result) => {
+        if(result) {
```

Comment:
I really not sure how to detect failed state for reset operation
https://github.com/brave/brave-core/pull/19212/commits/414fb6cd24fb8e0b6dfbf0eefabbcb9392065272#diff-ee2c62c697485a604bf332390aaf42dff62d59d4fdf5d5c24f8a75b9582c5082R128-R138
what we do it is:
- `Show Leo icon in the toolbar` -> On 
- `Show suggested prompts in the conversation` -> Off
- Change disclaimer to be shown when opening Ai Chat `brave.ai_chat.has_seen_disclaimer` -> false

at the end I just return `true` and as result show the alert with message `Leo was reset`
May be we don't need the alert message at all?

Reply:
Sorry, when I read this, I assumed the reset could fail because we were checking the value of result. As it can't fail, we an probably drop this result entirely :smile: